applications/crossbar/src/modules/cb_notifications.erl

```diff
@@ -462,11 +457,8 @@ maybe_delete_template(Context, Id, ContentType, TemplateJObj) ->
     case wh_doc:attachment(TemplateJObj, AttachmentName) of
         'undefined' ->
             lager:debug("failed to find attachment ~s", [AttachmentName]),
-            cb_context:add_system_error(
-              'bad_identifier'
-              ,wh_json:from_list([{<<"cause">>, ContentType}])
-              , Context
-             );
+            JObj = wh_json:from_list([{<<"cause">>, ContentType}]),
+            cb_context:add_system_error('bad_identifier', JObj, ContentType);
```

Comment:
Then you should include these in the plt?
Would it work that way?

> On 14 Oct 2015, at 12:01, James Aimonetti <notifications@github.com> wrote:
> 
> In applications/crossbar/src/modules/cb_notifications.erl:
> 
> > @@ -462,11 +457,8 @@ maybe_delete_template(Context, Id, ContentType, TemplateJObj) ->
> >      case wh_doc:attachment(TemplateJObj, AttachmentName) of
> >          'undefined' ->
> >              lager:debug("failed to find attachment ~s", [AttachmentName]),
> > -            cb_context:add_system_error(
> > -              'bad_identifier'
> > -              ,wh_json:from_list([{<<"cause">>, ContentType}])
> > -              , Context
> > -             );
> > +            JObj = wh_json:from_list([{<<"cause">>, ContentType}]),
> > +            cb_context:add_system_error('bad_identifier', JObj, ContentType);
> Dialyzer caught it for me. I suspect the way Travis/you run Dialyzer didn't include cb_context.beam so Dialyzer would not have known what to expect in that arg. I typically run it locally with:
> 
> dialyzer --plt .kazoo.plt file/i/changed.beam files/it/relies/on.beam
> So if my changeset in foo.erl calls bar:fun(X, Y), I'll include foo.beam and bar.beam. Plus I typically include wh_util, wh_json and props since most modules make some use of them.
> 
> —
> Reply to this email directly or view it on GitHub.
>